### PR TITLE
New PR template

### PR DESCRIPTION
**Pull request by yadudoc**
This PR adds a template for pull requests similar to the templates we have for issues. The goal is to make sure the PR nudges contributors to write cleaner PR descriptions that can go into our changelog.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+Please include a summary of the change and (optionally) which issue is fixed. Please also include
+relevant motivation and context.
+
+Fixes # (issue)
+
+## Type of change
+
+Please check the options below:
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
```
Comment (danielskatz) [inline .github/PULL_REQUEST_TEMPLATE.md:12]:
```suggestion
- [ ] Bug fix (non-breaking change that fixes an issue)
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+- [ ] New feature (non-breaking change which adds functionality)
```
Comment (danielskatz) [inline .github/PULL_REQUEST_TEMPLATE.md:13]:
```suggestion
- [ ] New feature (non-breaking change that adds functionality)
```

Comment (danielskatz) [reply .github/PULL_REQUEST_TEMPLATE.md:13]: you don't like that instead?

Comment (yadudoc) [reply .github/PULL_REQUEST_TEMPLATE.md]: Fixed.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Description
```
Comment (benclifford) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Be explicit about *who* the reader is.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+relevant motivation and context.
+
+Fixes # (issue)
+
+## Type of change
+
+Please check the options below:
+
+- [ ] Bug fix (non-breaking change that fixes an issue)
+- [ ] New feature (non-breaking change that adds functionality)
```
Comment (benclifford) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: should i be checking "new feature" and "breaking change" if i add a new feature that breaks things? if so, remove "non-breaking change" here and line above. if not, then split "breaking change" into breaking bugfix/breaking feature

**Pull request merged**